hadoop-hdfs-project/hadoop-hdfs-httpfs/src/main/java/org/apache/hadoop/fs/http/server/FSOperations.java

```diff
@@ -280,14 +279,14 @@ private static Map contentSummaryToJSON(ContentSummary contentSummary) {
    */
   @SuppressWarnings({"unchecked"})
   private static Map quotaUsageToJSON(QuotaUsage quotaUsage) {
-    Map response = new LinkedHashMap();
+    Map response = new TreeMap();
     Map quotaUsageMap = quotaUsageToMap(quotaUsage);
     response.put(HttpFSFileSystem.QUOTA_USAGE_JSON, quotaUsageMap);
     return response;
   }
 
   private static Map<String, Object> quotaUsageToMap(QuotaUsage quotaUsage) {
-    Map<String, Object> result = new LinkedHashMap<>();
+    Map<String, Object> result = new TreeMap<>();
```

Comment:
It would be nice to add a unit test to make sure that the order is preserved.

Reply:
I haven't found any existing unit tests that compares the order of the response JSON of WebHDFS and HttpFS. e.g. `BaseTestHttpFSWith#testQuotaUsage` checks the the correctness of output from DFS/HttpFS but doesn't care about the order.
That might be something we want to do with a new unit test class. Or is there one test class that does this already?

Initially I was just looking at output for LISTSTATUS. I discovered that WebHDFS responds with sorted keys, but HttpFS doesn't. Then I found other responses from HttpFS also aren't sorted but WebHDFS does. So I extended the scope of this jira.

Reply:
I honestly don't think people care about the order.

Reply:
You are right. People won't care in production.
I filed this jira just because I was comparing the RAW JSON.